File: kotlin/PlanesAndroid/app/src/test/java/VectorIteratorTest.java
import com.google.common.truth.Truth;
import com.planes.common.VectorIterator;

import org.junit.Test;

import java.util.Vector;

import static com.google.common.truth.Truth.assertThat;

public class VectorIteratorTest {
    @Test
    public void VectorIterator_Iterate() {
        VectorIterator<Integer> testVectorIterator = new VectorIterator<Integer>();
        Vector<Integer> testVector = new Vector<Integer>();
        testVector.add(1);
        testVector.add(2);
        testVector.add(3);
        testVectorIterator.setInternalList(testVector);

        int count = 0;
        while (testVectorIterator.hasNext()) {
            count++;
            assertThat(testVectorIterator.next() == count).isTrue();
        }
    }
}

File: kotlin/PlanesAndroid/app/src/test/java/VectorIteratorTest.kt
import com.google.common.truth.Truth
import com.planes_multiplayer.single_player_engine.VectorIterator
import org.junit.Test
import java.util.*

class VectorIteratorTest {
    @Test
    fun VectorIterator_Iterate() {
        val testVectorIterator = VectorIterator<Int>()
        val testVector = Vector<Int>()
        testVector.add(1)
        testVector.add(2)
        testVector.add(3)
        testVectorIterator.setInternalList(testVector)
        var count = 0
        while (testVectorIterator.hasNext()) {
            count++
            Truth.assertThat(testVectorIterator.next() == count).isTrue()
        }
    }
}